librz/core/cmd/cmd_analysis.c

```diff
@@ -3942,131 +3841,118 @@ static void cmd_sdbk(Sdb *db, const char *input) {
 	}
 }
 
-static void cmd_analysis_syscall(RzCore *core, const char *input) {
-	PJ *pj = NULL;
-	RzSyscallItem *si;
+RZ_IPI RzCmdStatus rz_analysis_syscall_show_handler(RzCore *core, int argc, const char **argv) {
+	st64 n = argc > 1 ? rz_num_math(core->num, argv[1]) : -1;
+	char *sysc = rz_core_syscall_as_string(core, n, core->offset);
+	if (!sysc) {
+		RZ_LOG_ERROR("Cannot resolve syscall: %" PFMT64d "\n", n);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println(sysc);
+	free(sysc);
+	return RZ_CMD_STATUS_OK;
+}
+
+static const char *syscallNumber(int n) {
+	return sdb_fmt(n > 1000 ? "0x%x" : "%d", n);
+}
+
+RZ_IPI RzCmdStatus rz_analysis_syscall_print_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
 	RzListIter *iter;
-	RzList *list;
-	RzNum *num = NULL;
-	int n;
+	RzSyscallItem *si;
+	RzList *list = rz_syscall_list(core->analysis->syscall);
```

Comment:
move this to a separate function in csyscall.c imo. you avoid having syscallNumber function both here and there. Also, move the other function that uses syscallNumber there.

Reply:
`syscallNumber()` is used in multiple places and is short enough to be duplicated. Further refactoring is out of the scope of this PR, IMHO. A lot of things has to be changed in how syscall are handled.